src/components/WelcomeScreen.tsx

```diff
@@ -74,13 +88,107 @@ export const WelcomeScreen: React.FC<Props> = ({
       );
     }
 
-    if (isWeb3Call) {
+    if (isWeb3Call && web3Account === null) {
+      const popupContainerStyle: React.CSSProperties = {
+        position: "fixed",
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0,
+        background: "rgba(0, 0, 0, 0.5)",
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        zIndex: 9999,
+      };
+      const popupContentStyle: React.CSSProperties = {
+        background: "white",
+        padding: "20px",
+        borderRadius: "5px",
+        textAlign: "center",
+      };
+
+      const buttonContainerStyle: React.CSSProperties = {
+        marginTop: "20px",
+        display: "flex",
+        justifyContent: "center",
+      };
+
+      const buttonStyle: React.CSSProperties = {
+        margin: "0 10px",
+        padding: "10px 20px",
+        border: "solid 1px #ccc",
+        borderRadius: "10px",
+        backgroundColor: "transparent",
+        color: "black",
+        cursor: "pointer",
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        overflow: "hidden",
+        flex: 1,
+      };
+
+      const buttonWrapperStyle: React.CSSProperties = {
+        display: "flex",
+        width: "100%",
+      };
+
+      const logoStyle: React.CSSProperties = {
+        width: "50px",
+        height: "50px",
+        marginBottom: "10px",
+        borderRadius: "50%",
+      };
+
+      return (
+        <div style={popupContainerStyle}>
```

Comment:
Please use the `css` prop provided by @emotion/react rather than the raw `style` prop. Why? Because emotion creates a css class and applies that rather than applying the styles directly to the element.

In this case it doesn't make much difference, other than for consistency, but in other cases especially for repeated elements, this can impact rendering time and page size significantly.